### ISSUE TITLE
Failure when inserting into the resource table.

### ISSUE THREAD

**Issue by rpfernandezjr** [tl-its-umich-edu/my-learning-analytics#829]
## Expected behavior (A clear and concise description of what you expected to happen) : 
When the cronjob runs the query against Big Query, The `resource` table should be populated with the needed values.

## Describe the bug (Tell us what happens instead of the expected behavior) :
When the cronjob runs, the following error is being thrown when attempting to insert data into the `resource` table:

```
[2019-12-17 16:22:47 -0500] [1847] [ERROR] Error running to_sql on table resource
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/sqlalchemy/engine/base.py", line 1216, in _execute_context
    cursor, statement, parameters, context
  File "/usr/local/lib/python3.6/site-packages/sqlalchemy/dialects/mysql/mysqldb.py", line 107, in do_executemany
    rowcount = cursor.executemany(statement, parameters)
  File "/usr/local/lib/python3.6/site-packages/MySQLdb/cursors.py", line 285, in executemany
    self._get_db().encoding)
  File "/usr/local/lib/python3.6/site-packages/MySQLdb/cursors.py", line 322, in _do_execute_many
    rows += self.execute(sql + postfix)
  File "/usr/local/lib/python3.6/site-packages/MySQLdb/cursors.py", line 255, in execute
    self.errorhandler(self, exc, value)
  File "/usr/local/lib/python3.6/site-packages/MySQLdb/connections.py", line 50, in defaulterrorhandler
    raise errorvalue
  File "/usr/local/lib/python3.6/site-packages/MySQLdb/cursors.py", line 252, in execute
    res = self._query(query)
  File "/usr/local/lib/python3.6/site-packages/MySQLdb/cursors.py", line 378, in _query
    db.query(q)
  File "/usr/local/lib/python3.6/site-packages/MySQLdb/connections.py", line 280, in query
    _mysql.connection.query(self, query)
_mysql_exceptions.OperationalError: (1048, "Column 'name' cannot be null")

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/code/dashboard/cron.py", line 309, in update_with_bq_access
    resource_df.to_sql(con=engine, name='resource', if_exists='append', index=False)
  File "/usr/local/lib/python3.6/site-packages/pandas/core/generic.py", line 2130, in to_sql
    dtype=dtype)
  File "/usr/local/lib/python3.6/site-packages/pandas/io/sql.py", line 450, in to_sql
    chunksize=chunksize, dtype=dtype)
  File "/usr/local/lib/python3.6/site-packages/pandas/io/sql.py", line 1127, in to_sql
    table.insert(chunksize)
  File "/usr/local/lib/python3.6/site-packages/pandas/io/sql.py", line 641, in insert
    self._execute_insert(conn, keys, chunk_iter)
  File "/usr/local/lib/python3.6/site-packages/pandas/io/sql.py", line 616, in _execute_insert
    conn.execute(self.insert_statement(), data)
  File "/usr/local/lib/python3.6/site-packages/sqlalchemy/engine/base.py", line 980, in execute
    return meth(self, multiparams, params)
  File "/usr/local/lib/python3.6/site-packages/sqlalchemy/sql/elements.py", line 273, in _execute_on_connection
    return connection._execute_clauseelement(self, multiparams, params)
  File "/usr/local/lib/python3.6/site-packages/sqlalchemy/engine/base.py", line 1099, in _execute_clauseelement
    distilled_params,
  File "/usr/local/lib/python3.6/site-packages/sqlalchemy/engine/base.py", line 1240, in _execute_context
    e, statement, parameters, cursor, context
  File "/usr/local/lib/python3.6/site-packages/sqlalchemy/engine/base.py", line 1458, in _handle_dbapi_exception
    util.raise_from_cause(sqlalchemy_exception, exc_info)
  File "/usr/local/lib/python3.6/site-packages/sqlalchemy/util/compat.py", line 296, in raise_from_cause
    reraise(type(exception), exception, tb=exc_tb, cause=cause)
  File "/usr/local/lib/python3.6/site-packages/sqlalchemy/util/compat.py", line 276, in reraise
    raise value.with_traceback(tb)
  File "/usr/local/lib/python3.6/site-packages/sqlalchemy/engine/base.py", line 1216, in _execute_context
    cursor, statement, parameters, context
  File "/usr/local/lib/python3.6/site-packages/sqlalchemy/dialects/mysql/mysqldb.py", line 107, in do_executemany
    rowcount = cursor.executemany(statement, parameters)
  File "/usr/local/lib/python3.6/site-packages/MySQLdb/cursors.py", line 285, in executemany
    self._get_db().encoding)
  File "/usr/local/lib/python3.6/site-packages/MySQLdb/cursors.py", line 322, in _do_execute_many
    rows += self.execute(sql + postfix)
  File "/usr/local/lib/python3.6/site-packages/MySQLdb/cursors.py", line 255, in execute
    self.errorhandler(self, exc, value)
  File "/usr/local/lib/python3.6/site-packages/MySQLdb/connections.py", line 50, in defaulterrorhandler
    raise errorvalue
  File "/usr/local/lib/python3.6/site-packages/MySQLdb/cursors.py", line 252, in execute
    res = self._query(query)
  File "/usr/local/lib/python3.6/site-packages/MySQLdb/cursors.py", line 378, in _query
    db.query(q)
  File "/usr/local/lib/python3.6/site-packages/MySQLdb/connections.py", line 280, in query
    _mysql.connection.query(self, query)
sqlalchemy.exc.OperationalError: (_mysql_exceptions.OperationalError) (1048, "Column 'name' cannot be null") [SQL: 'INSERT INTO resource (resource_type, id, course_id, name) VALUES (%s, %s, %s, %s)'] [parameters: (('canvas', '43560000008035244', 43560000000132242, 'IntroToPsychBanner.jpg'), ('canvas', '43560000009852687', 43560000000137876, 'Unit 14 C& D PPT  Student Fall 2019.pdf'), ('canvas', '43560000008215009', 43560000000127300, 'DES1111cpsSyllabus Fall 2019.pdf'), ('canvas', '43560000008264932', 43560000000127300, 'Instructor Meetings F19.dotx'), ('canvas', '43560000009560789', 43560000000138049, 'Study Sheet Final.pdf'), ('canvas', '43560000008233754', 43560000000137737, 'OLPD2811_MasterSyllabus_Fall2019.pdf'), ('canvas', '43560000009837699', 43560000000127300, 'DSD8 Convince F19.dotx'), ('canvas', '43560000007260332', 43560000000127300, 'textswapper2.html')  ... displaying 10 of 260 total bound parameter sets ...  ('canvas', '43560000008551375', 43560000000132242, 'Psych Tutoring Fall 2019 (1).pdf'), ('canvas', '43560000008101137', 43560000000121432, 'Cultural Dimensions & Engagement Reflection.docx'))] (Background on this error at: http://sqlalche.me/e/e3q8)
[2019-12-17 16:22:47 -0500] [1847] [INFO] (_mysql_exceptions.OperationalError) (1048, "Column 'name' cannot be null") [SQL: 'INSERT INTO resource (resource_type, id, course_id, name) VALUES (%s, %s, %s, %s)'] [parameters: (('canvas', '43560000008035244', 43560000000132242, 'IntroToPsychBanner.jpg'), ('canvas', '43560000009852687', 43560000000137876, 'Unit 14 C& D PPT  Student Fall 2019.pdf'), ('canvas', '43560000008215009', 43560000000127300, 'DES1111cpsSyllabus Fall 2019.pdf'), ('canvas', '43560000008264932', 43560000000127300, 'Instructor Meetings F19.dotx'), ('canvas', '43560000009560789', 43560000000138049, 'Study Sheet Final.pdf'), ('canvas', '43560000008233754', 43560000000137737, 'OLPD2811_MasterSyllabus_Fall2019.pdf'), ('canvas', '43560000009837699', 43560000000127300, 'DSD8 Convince F19.dotx'), ('canvas', '43560000007260332', 43560000000127300, 'textswapper2.html')  ... displaying 10 of 260 total bound parameter sets ...  ('canvas', '43560000008551375', 43560000000132242, 'Psych Tutoring Fall 2019 (1).pdf'), ('canvas', '43560000008101137', 43560000000121432, 'Cultural Dimensions & Engagement Reflection.docx'))] (Background on this error at: http://sqlalche.me/e/e3q8)
```

## Steps to Reproduce :
1. Run the cronjob `python manage.py runcrons --force`

## Additional context (Add any other context about the problem here) :
full log file of cron output: cron.txt
line throwing the error: https://github.com/tl-its-umich-edu/my-learning-analytics/blob/master/dashboard/cron.py#L303-L307

**Comment by jonespm** [comment]
This was fixed by #831

**Comment by jonespm** [comment]
This is working as expected with the improved cron (confirmed by Unizin)
